src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -205,10 +235,75 @@ export class ExamExerciseImportComponent implements OnInit {
             // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
             exercise.shortName?.length! > 2 &&
             this.shortNamePattern.test(exercise.shortName!) &&
+            !this.exercisesWithDuplicatedShortNames.has(exercise.id!) &&
             exercise.shortName !== this.getBlocklistShortNameOfProgrammingExercise(exercise.id!)
         );
     }
 
+    /**
+     * checks if there are any selected exercises with the same title or short name as the passed exercise
+     * @param exercise     exercise we want to use to check for duplications
+     * @param exerciseGroup exercise group of the exercise
+     * @param checkForTitle true if the title should be checked, otherwise the short name is checked
+     */
+    checkForDuplicatedTitlesOrShortNames(exercise: Exercise, exerciseGroup: ExerciseGroup, checkForTitle: boolean) {
+        if (!this.exerciseIsSelected(exercise, exerciseGroup)) {
+            return;
+        }
+
+        const duplicatesToCheck = checkForTitle ? this.exercisesWithDuplicatedTitles : this.exercisesWithDuplicatedShortNames;
+        let hasDuplicate = false;
+        this.selectedExercises.forEach((exerciseGroup) => {
+            exerciseGroup.forEach((ex) => {
+                if (ex.type === ExerciseType.PROGRAMMING && ex !== exercise) {
+                    if ((checkForTitle && ex.title === exercise.title) || (!checkForTitle && ex.shortName === exercise.shortName)) {
+                        hasDuplicate = true;
+                        duplicatesToCheck.set(ex.id!, checkForTitle ? exercise.title! : exercise.shortName!);
+                    }
+                }
+            });
+        });
```

Comment:
```suggestion
        const hasDuplicate = this.selectedExercises.some((exerciseGroup) => {
            for (let ex of exerciseGroup) {
                if (ex.type === ExerciseType.PROGRAMMING && ex !== exercise) {
                    if ((checkForTitle && ex.title === exercise.title) || (!checkForTitle && ex.shortName === exercise.shortName)) {
                        duplicatesToCheck.set(ex.id!, checkForTitle ? exercise.title! : exercise.shortName!);
                        return true;
                    }
                }
                return false;
            }
        });
```

Reply:
I tried using `some` but the problem is that not all duplicates are detected. It returns `true` as soon as one duplicate has been found instead of looking through the whole set and adding all duplicates to the `duplicatesToCheck`. This means in the end the exercise that is being checked and one other exercise get added, all other possible duplicates get skipped.

Reply:
I see 👍